### DISCUSSION TITLE
i3 does not respect xmodmap changes consistently?

### DISCUSSION THREAD

**Discussion by GeekyPeas** [i3/i3#5128] [Q&A] [answered]
One of my issues got closed by the bot. I am new here and may be I did not format the post correctly or something (I am not sure what!). But that is besides the point for now. I want help with the following:

## Issue Description:
The [keyindicator](https://github.com/vivien/i3blocks-contrib/tree/master/keyindicator) for `Caps Lock` breaks if I use `xmodmap` to change `Caps_Lock` to `Escape` and `Shift+Caps_Lock` to `Caps_Lock`.

In particular this line breaks:

    bindsym --release Caps_Lock exec pkill -SIGRTMIN+11 i3blocks

And this line restores expected behavior:

    bindsym --release Shift+Caps_Lock exec pkill -SIGRTMIN+11 i3blocks

So I have prepared the following _minimum reproducible bug_ to explain what I mean:
**EDIT:** I have replaced the original _minimum_reproducible_bug_ with a cleaner version below:

Take this as your `i3/config`:

    bindsym $mod+Escape exec notify-send "mod + Escape pressed"
    bindsym $mod+Caps_Lock exec notify-send "mod + Caps_Lock pressed"
    bindsym --release Caps_Lock exec notify-send "Caps_Lock released"
    bindsym --release Shift+Caps_Lock exec notify-send "Shift+Caps_Lock released"

Without any `xmodmap` modifications (that is, with all defaults) everything should work as expected, that is:

1. Pressing `$mod+Escape` will give notification "mod + Escape pressed"
2. Pressing `$mod+Caps_Lock` will give notification "mod + Caps_Lock pressed"
3. Pressing and releasing `Caps_Lock` will give notification "Caps_Lock released"
4. Pressing and releasing `Shift+Caps_Lock` will give notification "Shift+Caps_Lock released"

Now apply the `xmodmap` modifications with `xmodmap -e 'clear Lock'; xmodmap -e "keycode 66 = Escape Caps_Lock" `. With these modifications: 

1. Pressing `$mod+Escape` will give notification "mod + Escape pressed"
2. Pressing `$mod+Caps_Lock` will give notification "mod + Escape pressed" (That is `Caps_Lock` is interpreted as `Escape`).
3. Pressing and releasing `Caps_Lock` will give notification "Caps_Lock released" (That is `Caps_Lock` is interpreted as `Caps_Lock`).
4. Pressing and releasing `Shift+Caps_Lock` will give notification "Shift+Caps_Lock released"

I feel `Caps_Lock` being interpreted as two different things is inconsistent.

So summary:
1. `$mod+key` is interpreted as per `xmodmap` changes
2. `key` is interpreted as per defaults (without any `xmodmap` changes)

So my **questions** is: I want `i3` to sensibly and consistently respect changes applied through `xmodmap`. Is there a way to do that?

**Comment by orestisfl** [comment, marked as answer]
The reason why the bot closed your issue is explained here https://github.com/i3/i3/issues/5127#issuecomment-1249125110

Your i3 version is older than the latest release.

But since I can still reproduce the issue, I'll reopen it and favor the issue over this discussion which I will lock.
